### PR TITLE
Fill table 1

### PR DESCRIPTION
**Pull request by Fr0stmourne**

### CODE REVIEW

File: src/js/NasaTable/NasaTable.js [hunk 1]
```diff
@@ -48,46 +62,70 @@ class NasaTable extends Component {
...
+  setDataToShow = ({ rowsPerPage, page, data, sortKey, isDescendingSort }) => {
+    const from = rowsPerPage * page;
+    const visibleRows = this.getDataToShow(data.table, from, rowsPerPage, sortKey, isDescendingSort);
+    this.setState({ data, visibleRows });
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.isDescendingSort !== this.state.isDescendingSort) {
+      const { rowsPerPage, page, data, sortKey, isDescendingSort } = this.state;
+      this.setDataToShow({ rowsPerPage, page, data, sortKey, isDescendingSort });
```
Comment (Fr0stmourne) [inline src/js/NasaTable/NasaTable.js:73]: is it OK to write like that for better readability (all the function arguments are clear) or should I use just ` this.setDataToShow(this.state)`?

File: src/js/Header/Header.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import Typography from '@material-ui/core/Typography';
 import IconButton from '@material-ui/core/IconButton';
 import MenuIcon from '@material-ui/icons/Menu';
 
+import './header.scss';
+
 const Header = () => (
   <AppBar position="sticky" className="table-header">
```
Comment (Ivanekrasov) [inline src/js/Header/Header.js]: Move inline style to css file please

File: src/js/ImageDialog/ImageDialog.js [hunk 3]
```diff
@@ -4,26 +4,21 @@ import DialogActions from '@material-ui/core/DialogActions';
...
-        <DialogTitle id="alert-dialog-title">Mars</DialogTitle>
+        <DialogTitle id="alert-dialog-title">{imageName}</DialogTitle>
         <DialogContent>
-          <img
-            src="https://upload.wikimedia.org/wikipedia/commons/0/02/OSIRIS_Mars_true_color.jpg"
-            alt="zaglushka"
-            width="255"
-            height="255"
-          />
+          <img src={image} alt="zaglushka" width="255" height="255" />
```
Comment (Ivanekrasov) [inline src/js/ImageDialog/ImageDialog.js]: These too

File: src/js/NasaTable/NasaTable.js [hunk 4]
```diff
@@ -6,8 +6,15 @@ import TableHead from '@material-ui/core/TableHead';
 import TableRow from '@material-ui/core/TableRow';
 import TablePagination from '@material-ui/core/TablePagination';
 
-import ImageDialog from './ImageDialog';
-import { getNasaData } from '../api/table';
+import ImageDialog from '../ImageDialog';
+
```
Comment (Ivanekrasov) [inline src/js/NasaTable/NasaTable.js]: Extra empty line

File: src/js/NasaTable/NasaTable.js [hunk 5]
```diff
@@ -48,46 +62,70 @@ class NasaTable extends Component {
...
               {headers &&
                 headers.map((el, i) => (
-                  <TableCell className="table-headers" key={i}>
+                  <TableCell
+                    className="table-headers"
+                    key={i}
+                    onClick={() =>
+                      isDescendingSort
+                        ? this.setState({ isDescendingSort: false, sortKey: headersMapping[el] })
+                        : this.setState({ isDescendingSort: true, sortKey: headersMapping[el] })
```
Comment (Ivanekrasov) [inline src/js/NasaTable/NasaTable.js]: You'd better move it to a separate function

File: src/js/NasaTable/NasaTable.js (hunk 1 above)
Comment (Ivanekrasov) [reply src/js/NasaTable/NasaTable.js:73]: If you need all values from the state you can use spread operator (e.g. `{ ...this.state }`). This is an equivalent to Object.assign. But you should be aware of using only primitive values here - Object.assign makes only a shallow copy

**Review by Ivanekrasov — CHANGES_REQUESTED** [review]

**Pull request merged**